librz/core/cmd/cmd_cmp.c

```diff
@@ -234,237 +76,25 @@ static int rizin_compare_unified(RzCore *core, ut64 of, ut64 od, int len) {
 	return true;
 }
 
-static int rizin_compare(RzCore *core, const ut8 *f, const ut8 *d, int len, int mode) {
-	int i, eq = 0;
-	PJ *pj = NULL;
-	if (len < 1) {
-		return 0;
-	}
-	if (mode == 'j') {
-		pj = pj_new();
-		if (!pj) {
-			return -1;
-		}
-		pj_o(pj);
-		pj_k(pj, "diff_bytes");
-		pj_a(pj);
-	}
-	for (i = 0; i < len; i++) {
-		if (f[i] == d[i]) {
-			eq++;
-			continue;
-		}
-		switch (mode) {
-		case 0:
-			rz_cons_printf("0x%08" PFMT64x " (byte=%.2d)   %02x '%c'  ->  %02x '%c'\n",
-				core->offset + i, i + 1,
-				f[i], (IS_PRINTABLE(f[i])) ? f[i] : ' ',
-				d[i], (IS_PRINTABLE(d[i])) ? d[i] : ' ');
-			break;
-		case '*':
-			rz_cons_printf("wx %02x @ 0x%08" PFMT64x "\n",
-				d[i],
-				core->offset + i);
-			break;
-		case 'j':
-			pj_o(pj);
-			pj_kn(pj, "offset", core->offset + i);
-			pj_ki(pj, "rel_offset", i);
-			pj_ki(pj, "value", (int)f[i]);
-			pj_ki(pj, "cmp_value", (int)d[i]);
-			pj_end(pj);
-			break;
-		}
-	}
-	if (mode == 0) {
-		eprintf("Compare %d/%d equal bytes (%d%%)\n", eq, len, (eq / len) * 100);
```

Comment:
Changing this `eprintf` to `rz_cons_printf` in `rz_cmp_print` results in this line being printed the last. I would want it to print before the rest of the lines. Any way to do it without storing the diffs in a buffer, and printing them in the end?

Reply:
Were you able to solve this?